library/alloc/src/boxed.rs

```diff
@@ -1259,8 +1150,9 @@ impl<T: ?Sized, A: Allocator> Box<T, A> {
     #[stable(feature = "box_raw", since = "1.4.0")]
     #[inline]
     pub fn into_raw(b: Self) -> *mut T {
-        // Make sure Miri realizes that we transition from a noalias pointer to a raw pointer here.
-        unsafe { &raw mut *&mut *Self::into_raw_with_allocator(b).0 }
+        // Avoid `into_raw_with_allocator` as that interacts poorly with Miri's Stacked Borrows.
```

Comment:
@RalfJung the old comment gave some explanation for the `&raw mut` dance that might be good to preserve.

Reply:
We're not doing the dance any more -- the old comment referred to the `&mut` and that is gone.

Reply:
That said, I can still extend the comment to point out the main subtle point going on here -- I just pushed that.

Reply:
Ah, fair enough.